src/components/table/table.tsx

```diff
@@ -113,16 +128,29 @@ export function Table<T extends object>({
           [state, controlledState]
         )
       },
-      groupByFn,
+      toggleSelectedItemClb,
+      itemIsDisabled,
     },
     ...reactTableHooks
   )
 
   useEffect(() => {
-    if (selectedItemsClb) {
-      selectedItemsClb(selectedFlatRows.map((r: Row<T>) => r.original))
+    if ((selectedFlatRows.length === 0 || isAllRowsSelected) && selectedItemsClb) {
+      const isGrouped = groupBy.length > 0
+      selectedItemsClb(
+        selectedFlatRows.reduce((h: Item[], r: Row<T>) => {
+          if (isGrouped && r.isGrouped) return h
+          if (itemIsDisabled(r)) return h
+          return [...h, r.original]
```

Comment:
Nothing wrong with this, however it's safe to use `push` in potentially lengthy iterations, to not re-create the array. I think it would be faster, yet less idiomatic.